src/NServiceBus.Core/TaskEx.cs

```diff
@@ -16,5 +17,25 @@ public static void Ignore(this Task task)
 
         public static readonly Task<bool> TrueTask = Task.FromResult(true);
         public static readonly Task<bool> FalseTask = Task.FromResult(false);
+
+        public static Task<T> ThrowIfNull<T>(this Task<T> task, string methodName)
```

Comment:
This (and the overload) can be moved to another class if having it in TaskEx. I felt like it was the best fit.